src/object/cli_ec.c

```diff
@@ -1339,6 +1353,155 @@ obj_ec_req_reasb(daos_obj_rw_t *args, daos_obj_id_t oid,
 	return rc;
 }
 
+static struct obj_ec_recov *
+obj_ec_recov_alloc(struct daos_oclass_attr *oca)
+{
+	struct obj_ec_recov	*recov;
+	unsigned short		 k = obj_ec_data_tgt_nr(oca);
+	unsigned short		 p = obj_ec_parity_tgt_nr(oca);
+	void			*buf, *tmp_ptr;
+	size_t			 struct_size, tbl_size, matrix_size;
+	size_t			 idx_size, list_size, err_size;
+
+	struct_size = roundup(sizeof(struct obj_ec_recov), 8);
+	tbl_size = k * p * 32;
+	matrix_size = roundup((k + p) * k, 8);
+	idx_size = roundup(sizeof(uint32_t) * k, 8);
+	list_size = roundup(sizeof(uint32_t) * p, 8);
+	err_size = roundup(sizeof(bool) * (k + p), 8);
+
+	D_ALLOC(buf, struct_size + tbl_size + 3 * matrix_size + idx_size +
+		     list_size + err_size);
```

Comment:
we are doing this because we want to reduce malloc calls right?

Reply:
right, there are some other places also with similar usage.